src/ansys/fluent/core/session_meshing.py

```diff
@@ -0,0 +1,202 @@
+        return self._meshing
+
+    @property
+    def workflow(self):
+        """workflow datamodel root."""
+        if self.solver_switch:
+            raise AttributeError(
+                "Mesh-Session-specific attributes are not available in Solver-Session"
+            )
+        if self._workflow is None:
+            try:
+                from ansys.fluent.core.datamodel.workflow import Root as workflow_root
+
+                self._workflow = workflow_root(self._se_service, "workflow", [])
+            except (ImportError, ModuleNotFoundError):
+                LOG.warning(_CODEGEN_MSG_DATAMODEL)
+                self._workflow = PyMenuGeneric(self._se_service, "workflow")
+        return self._workflow
+
+    @property
+    def PartManagement(self):
+        """PartManagement datamodel root."""
+        if self.solver_switch:
+            raise AttributeError(
+                "Mesh-Session-specific attributes are not available in Solver-Session"
+            )
+        if self._part_management is None:
+            try:
+                from ansys.fluent.core.datamodel.PartManagement import (
+                    Root as PartManagement_root,
+                )
+
+                self._part_management = PartManagement_root(
+                    self._se_service, "PartManagement", []
+                )
+            except (ImportError, ModuleNotFoundError):
+                LOG.warning(_CODEGEN_MSG_DATAMODEL)
+                self._part_management = PyMenuGeneric(
+                    self._se_service, "PartManagement"
+                )
+        return self._part_management
+
+    @property
+    def PMFileManagement(self):
+        """PMFileManagement datamodel root."""
+        if self.solver_switch:
+            raise AttributeError(
+                "Mesh-Session-specific attributes are not available in Solver-Session"
+            )
+        if self._pm_file_management is None:
+            try:
+                from ansys.fluent.core.datamodel.PMFileManagement import (
+                    Root as PMFileManagement_root,
+                )
+
+                self._pm_file_management = PMFileManagement_root(
+                    self._se_service, "PMFileManagement", []
+                )
+            except (ImportError, ModuleNotFoundError):
+                LOG.warning(_CODEGEN_MSG_DATAMODEL)
+                self._pm_file_management = PyMenuGeneric(
+                    self._se_service, "PMFileManagement"
+                )
+        return self._pm_file_management
+
+    def switch_to_solver(self):
+        """A switch to move to the solver session from meshing."""
+        if self.solver_switch:
+            raise AttributeError(
+                "Mesh-Session-specific attributes are not available in Solver-Session"
+            )
+        self.tui.switch_to_solution_mode("yes")
+        solver_session = Solver(fluent_connection=self.fluent_connection)
+        self.solver_switch = True
```

Comment:
@prmukherj with this change, guarantee that the invalid attributes are unavailable immediately after switching the backend, even if the next line throws.
```suggestion
        self.solver_switch = True
        solver_session = Solver(fluent_connection=self.fluent_connection)
```